src/datasources/asset/AssetDataSource.ts

```diff
@@ -70,10 +96,203 @@ export class AssetDataSource extends DataSourceBase<AssetMetadataQuery> {
     return result;
   }
 
-  shouldRunQuery(_: AssetMetadataQuery): boolean {
+  async processUtilizationQuery(query: AssetUtilizationQuery, options: DataQueryRequest) {
+    const result: DataFrameDTO = { refId: query.refId, fields: [] };
+    let workspaceId = this.templateSrv.replace(query.workspace);
+    // fetch and process utilization raw data for chosen assets/systems
+    const utilization_array = await this.fetchAndProcessUtilizationData(query, options);
+    if (utilization_array.length) {
+      let utilization_array_with_alias: Array<{ id: string, datetimes: number[], values: number[], alias: string }>;
+      const idsArray = utilization_array.map((data) => {
+        return data.id
+      });
+      // find and add aliases to data
+      if (query.entityType === EntityType.System) {
+        const filterArr: string[] = [];
```

Comment:
Instead of if else if else, please use methods and functions . 

Make your code readable by keeping your methods small.